paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/TransformSpecToElement.kt

```diff
@@ -0,0 +1,90 @@
+package com.stripe.android.paymentsheet.forms
+
+import com.stripe.android.paymentsheet.FocusRequesterCount
+import com.stripe.android.paymentsheet.FormElement
+import com.stripe.android.paymentsheet.elements.EmailConfig
+import com.stripe.android.paymentsheet.elements.NameConfig
+import com.stripe.android.paymentsheet.elements.common.DropdownFieldController
+import com.stripe.android.paymentsheet.elements.common.TextFieldController
+import com.stripe.android.paymentsheet.elements.country.CountryConfig
+import com.stripe.android.paymentsheet.specifications.FormItemSpec
+import com.stripe.android.paymentsheet.specifications.LayoutSpec
+
+/**
+ * The purpose of this class is to transform a LayoutSpec data object into an Element, which
+ * has a controller and identifier.  With only a single field in a section the section
+ * controller will be a pass through of the field controller.
+ */
+internal class TransformSpecToElement {
```

Comment:
Would it make sense for this to be part of the FormElement class, as something like a `create` method?

Reply:
I would like to keep formElement more of a concrete class in that it doesn't need to know about Spec or be dependent on it.  Transform is then dependent on SPec and ELement and knows about both.